docs/style-spec/expressions.md

```diff
@@ -0,0 +1,40 @@
+# Style Function Expressions
+
+**Constants:**
+- LN2
+- PI
+- e
```

Comment:
I agree that we need to avoid a mini-language in the JSON format – that’s what derailed #4079 for so long.

My original comment above refers to an approach we could take for NSExpression conversion, since that DSL already makes a distinction between key paths, variables, and strings. For example, the iOS SDK has the following [guidance](https://www.mapbox.com/ios-sdk/api/3.5.2/Classes/MGLVectorStyleLayer.html#/c:objc%28cs%29MGLVectorStyleLayer%28py%29predicate) for using the special `$id` and `$type` feature properties in a predicate:

> It is possible to create expressions that contain special characters in the predicate format syntax. This includes the `$` in the `$id` and `$type` special style attributes and also `hyphen-minus` and `tag:subtag`. However, you must use `%K` in the format string to represent these variables: `@"%K == 'LineString'", @"$type"`.

In that example, the developer would want `$type` treated as a key path. If they put `$type` directly in the format string, NSPredicate interprets it as a variable instead. For these mathematical constants, however, the developer would expect it to be interpreted as a variable, making an inline `$pi` desirable.